// 错误代码和错误信息的集合
const errors = {
  'not_defined': '发生未知的错误',
  '400': '未登录',
  '401': '没有权限',
  '402': '登录超时',
  '403': '登录非法',
  '500': '服务器错误',
  '200': '正常返回',
  '600': '操作失败',
  '60101': '账号不存在',
  '60102': '用户名或者密码错误',
  '60103': '账号已存在',
  '60104': '验证码错误',
  '60105': '短信验证码发送失败',
  '60106': '记录已存在',
  '60107': '请勿重复申请同一家单位',
  '60108': '邮箱已存在',
  '60109': '手机已存在',
  '60110': '系统删除失败',
  '60111': '手机号输入不合法',
  '60112': '用户名输入不合法',
  '60113': '邮箱输入不合法',
  '60118': '信息已失效',
  '60119': '实名认证失败'
}

/**
 * 使用错误代码获取对应的错误提示信息
 * @param {Number | String} code 错误代码
 * @returns {String} 错误信息
 */
export function getErrorByCode (code) {
  if (errors.hasOwnProperty(code)) {
    for (let val in errors) {
      if (val === code) {
        return errors[val]
      }
    }
  } else {
    return '服务器出现未知错误'
  }
}
